src/Microsoft.Graph.Core/Requests/BaseClient.cs

```diff
@@ -31,6 +33,25 @@ public class BaseClient : IBaseClient
             this.HttpProvider = httpProvider ?? new HttpProvider(new Serializer());
         }
 
+        /// <summary>
+        /// Constructs a new <see cref="BaseClient"/>.
+        /// </summary>
+        /// <param name="baseUrl">The base service URL. For example, "https://graph.microsoft.com/v1.0."</param>
+        /// <param name="tokenCredential">The <see cref="TokenCredential"/> for authenticating request messages.</param>
+        /// <param name="scopes">List of scopes for the authentication context.</param>
+        /// <param name="httpProvider">The <see cref="IHttpProvider"/> for sending requests.</param>
+        public BaseClient(
+            string baseUrl,
+            TokenCredential tokenCredential,
+            IEnumerable<string> scopes = null,
+            IHttpProvider httpProvider = null
+            )
+        {
+            this.BaseUrl = baseUrl;
```

Comment:
no validation on the base url?

Reply:
This is already done [here ](https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/8c83235d0bca749a05a78b270cdf7e869a278c49/src/Microsoft.Graph.Core/Requests/BaseClient.cs#L79)on the property setter.